todoman/interactive.py

```diff
@@ -0,0 +1,583 @@
+        self.body = urwid.ListBox(urwid.SimpleFocusListWalker(items))
+        super().__init__(parent, callback)
+
+    def items_to_display(self):
+        '''
+        Create a list of TodomanItems to display, based on the associated
+        Database and the current done_is_hidden setting.
+
+        (TodomanItemListPage) -> [TodomanItem]
+        '''
+        items = []
+        for filename in self.database.todos.keys(): 
+            todo = TodomanItem(filename, self.database, self.generate_label)
+            if not self.done_is_hidden or not todo.is_completed:
+                items.append(todo)
+        items.sort(key = lambda item: item.label.lower())
+        return items
+
+    def callback_move_to(self, **kwargs):
+        '''
+        Move the TodomanItem in focus to Database database.
+
+        (TodomanItemListPage, Database) -> None
+        '''
+        for key, value in kwargs.items():
+            if key == "database" and value != self.database:
+                value.save(self.body.focus.todo)
+                self.database.delete(self.body.focus.todo)
+                self.body.body.remove(self.body.focus)
+                self.statusbar = "Item moved to {0}.".format(value.name)
+
+    def move_database_chooser(self):
+        '''
+        Open a TodomanDatabasesPage from which to choose the destination of the
+        move operation for the selected item.
+
+        (TodomanItemListPage) -> None
+        '''
+        newPage = TodomanDatabasesPage(self.parent, self.callback_move_to)
+        self.open_page(newPage)
+
+    def callback_copy_to(self, **kwargs):
+        '''
+        Copy the TodomanItem in focus to Database database.
+
+        (TodomanItemListPage, Database) -> None
+        '''
+        for key, value in kwargs.items():
+            if key == "database" and value != self.database:
+                value.save(self.body.focus.todo)
+                self.statusbar = "Item copied to {0}.".format(value.name)
+
+    def copy_database_chooser(self):
+        '''
+        Open a TodomanDatabasesPage from which to choose the destination of the
+        copy operation for the selected item.
+
+        (TodomanItemListPage) -> None
+        '''
+        newPage = TodomanDatabasesPage(self.parent, self.callback_copy_to)
+        self.open_page(newPage)
+
+    def delete(self, item = None):
+        '''
+        Delete the TodomanItem item from its database. By default, delete the
+        item in focus.
+
+        (TodomanItemListPage, TodomanItem) -> None
+        '''
+        if item == None:
+            item = self.body.focus
+        item.database.delete(item.todo)
+        self.body.body.remove(item)
+
+    def generate_label(self, item):
+        '''
+        Return a label for a given TodomanItem for display in the
+        TodomanItemListPage listing.
+
+        (TodomanItemListPage, TodomanItem) -> str
+        '''
+        return "{0} {1}".format('!' if item.has_priority else ' ',
```

Comment:
We already have `todoman.ui.TodoFormatter` for this (which will also add colors)